### PR TITLE
feat(docs): update Algolia configuration information

### PR DESCRIPTION
**Pull request by pepopowitz**
## Description

Addresses feedback from #2438.

Updates our documentation on Algolia DocSearch integration.

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [x] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: howtos/configuration.md [hunk 1]
```diff
@@ -10,9 +10,26 @@ The documentation can be customized by setting parameters in [docusaurus.config.
 
 ## Search by Algolia
 
-- API-Key and Index name is configured in `./docusaurus.config.js`.
+Internal search is handled by an integration with Algolia DocSearch.
```
Comment (christinaausley) [inline howtos/configuration.md:13]: Is it worth linking to [Algolia DocSearch](https://docsearch.algolia.com/) here?

File: howtos/configuration.md [hunk 2]
```diff
@@ -10,9 +10,26 @@ The documentation can be customized by setting parameters in [docusaurus.config.
...
 
-[Algolia DocSearch](https://docsearch.algolia.com/) configuration is **NOT** a part of this repository. Config can be found on a dedicated [Docsearch Configurations](https://github.com/algolia/docsearch-configs) repo owned by Algolia but maintained by the individual projects. You must prove you are associated with the project to modify the configuration in the [Camunda config](https://github.com/algolia/docsearch-configs/blob/master/configs/camunda.json).
+### Configuration
+
+Our Algolia API key, app ID, and index name are configured in `./docusaurus.config.js`.
+
+Configuration of Algolia's index and crawler can be edited at https://crawler.algolia.com/.
```
Comment (christinaausley) [inline howtos/configuration.md:19]: Configuration of Algolia's index and crawler can be edited with the [Algolia Crawler](https://crawler.algolia.com/), or something of the like?

File: howtos/configuration.md [hunk 3]
```diff
@@ -10,9 +10,26 @@ The documentation can be customized by setting parameters in [docusaurus.config.
...
+```
+      pathsToMatch: [
+        "https://docs.camunda.io/docs/**",
+        "https://docs.camunda.io/optimize/**",
+      ],
+```
+
+### Troubleshooting
+
+Algolia's index of our documentation can be explored directly at https://dashboard.algolia.com/.
```
Comment (christinaausley) [inline howtos/configuration.md:32]: Take a closer look at [Algolia's index of our documentation](https://dashboard.algolia.com/).

**Review by christinaausley — DISMISSED** [review]
A few non-blocking suggestions for linkage 👍

File: howtos/configuration.md (hunk 1 above)
Comment (pepopowitz) [reply howtos/configuration.md:13]:
```suggestion
Internal search is handled by an integration with [Algolia DocSearch](https://docsearch.algolia.com/).
```

File: howtos/configuration.md (hunk 2 above)
Comment (pepopowitz) [reply howtos/configuration.md:19]:
```suggestion
Configuration of Algolia's index and crawler can be edited with the [Algolia Crawler](https://crawler.algolia.com/).
```

File: howtos/configuration.md (hunk 3 above)
Comment (pepopowitz) [reply howtos/configuration.md:32]:
```suggestion
Take a closer look at [Algolia's index of our documentation](https://dashboard.algolia.com/).
```

**Comment by pepopowitz** [comment]
@christinaausley thanks for the suggestions! They've been applied.

**Review by christinaausley — APPROVED** [review]

**Review by akeller — APPROVED** [review]
Remember when the crawler code was on their public GitHub? 😆 API keys and all 💀

**Pull request merged**